File: C++/1701. Average Waiting Time.cpp
class Solution {
public:
    double averageWaitingTime(vector<vector<int>>& customers) {
        vector<int>v(customers.size(),0);
        if(customers.size()==1){
            return customers[0][1];
        }
        long long int ans=customers[0][0]+customers[0][1];
        v[0]=customers[0][1];
        long long int total=v[0];
        for(int i=1;i<customers.size();i++){
              long long int ans1;
            if(ans>customers[i][0])
            ans1=ans+customers[i][1];
            else
                ans1=customers[i][0]+customers[i][1];
            ans=ans1;
            if(ans1<customers[i][0]){
                v[i]=customers[i][1];
            }
            else
            v[i]=ans1-customers[i][0];
            total+=v[i];
        }
        return total*(1.0)/customers.size();
    }
};

File: JAVA/1701. Average Waiting Time.java
class Solution {
    public double averageWaitingTime(int[][] customers) {
        long time=customers[0][0];
        long w_time=0;
        for(int[] cus:customers){
            if(time<cus[0]){
                time=cus[0];
            }
            time+=cus[1];
            w_time+=time-cus[0];
        }
        return (double)w_time/customers.length;
    }
}
